include/k4arecord/record.hpp

```diff
@@ -0,0 +1,280 @@
+/** \file record.hpp
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * Kinect For Azure SDK - C++ wrapper.
+ */
+#ifndef K4A_RECORD_HPP
+#define K4A_RECORD_HPP
+
+#include <k4a/k4a.hpp>
+#include <k4arecord/record.h>
+
+namespace k4a
+{
+
+/** \class record record.hpp
+ * Wrapper for \ref k4a_record_t
+ *
+ * Wraps a handle for a record object
+ *
+ * \sa k4a_record_t
+ */
+class record
+{
+public:
+    /** Creates a k4a::record from a k4a_record_t
+     * Takes ownership of the handle, i.e. you should not call
+     * k4a_record_close on the handle after giving it to the
+     * k4a::record; the k4a::record will take care of that.
+     */
+    record(k4a_record_t handle = nullptr) noexcept : m_handle(handle) {}
+
+    /** Moves another k4a::record into a new k4a::record
+     */
+    record(record &&other) noexcept : m_handle(other.m_handle)
+    {
+        other.m_handle = nullptr;
+    }
+
+    record(const record &) = delete;
+
+    ~record()
+    {
+        flush();
+        close();
+    }
+
+    record &operator=(const record &) = delete;
+
+    /** Moves another k4a::record into this k4a::record; other is set to invalid
+     */
+    record &operator=(record &&other) noexcept
+    {
+        if (this != &other)
+        {
+            flush();
+            close();
+            m_handle = other.m_handle;
+            other.m_handle = nullptr;
+        }
+
+        return *this;
+    }
+
+    /** Returns true if the k4a::record is valid, false otherwise
+     */
+    operator bool() const noexcept
+    {
+        return m_handle != nullptr;
+    }
+
+    /** Closes a K4A recording.
+     *
+     * \sa k4a_record_close
+     */
+    void close() noexcept
+    {
+        if (m_handle)
+        {
+            k4a_record_close(m_handle);
+            m_handle = nullptr;
+        }
+    }
+
+    /** Flushes all pending recording data to disk
+     *
+     * \sa k4a_record_flush
+     */
+    void flush()
+    {
+        if (m_handle)
+        {
+            k4a_record_flush(m_handle);
```

Comment:
This function can fail, so we need a throw here if this fails.

Reply:
It is my mistake. Add throw an error.